src/iteminfo_query.h

```diff
@@ -0,0 +1,223 @@
+    GUN_CAPACITY,
+    GUN_TYPE,
+    GUN_MAGAZINE,
+
+    AMMO_REMAINING,
+    AMMO_UPSCOST,
+
+    GUN_MAX_RANGE,
+    GUN_AIMING_STATS,
+    GUN_DAMAGE,
+    GUN_DAMAGE_LOADEDAMMO,
+    GUN_DAMAGE_TOTAL,
+    GUN_ARMORPIERCE,
+    GUN_ARMORPIERCE_LOADEDAMMO,
+    GUN_ARMORPIERCE_TOTAL,
+    GUN_DISPERSION,
+    GUN_DISPERSION_LOADEDAMMO,
+    GUN_DISPERSION_TOTAL,
+    GUN_DISPERSION_SIGHT,
+
+    GUN_RECOIL,
+    GUN_RECOIL_BIPOD,
+
+    GUN_RECOMMENDED_STRENGTH,
+    GUN_RELOAD_TIME,
+
+    GUN_FIRE_MODES,
+    GUN_ALLOWED_MAGAZINES,
+
+    DESCRIPTION_GUN_MODS,
+    DESCRIPTION_GUN_CASINGS,
+
+
+    DESCRIPTION_GUNMOD,
+    DESCRIPTION_GUNMOD_REACH,
+
+    GUNMOD_DISPERSION,
+    GUNMOD_DISPERSION_SIGHT,
+    GUNMOD_AIMSPEED,
+    GUNMOD_DAMAGE,
+    GUNMOD_ARMORPIERCE,
+    GUNMOD_HANDLING,
+    GUNMOD_AMMO,
+
+    GUNMOD_USEDON,
+    GUNMOD_LOCATION,
+
+
+    ARMOR_BODYPARTS,
+    ARMOR_LAYER,
+    ARMOR_COVERAGE,
+    ARMOR_WARMTH,
+    ARMOR_ENCUMBRANCE,
+    ARMOR_STORAGE,
+    ARMOR_PROTECTION,
+
+
+    BOOK_SUMMARY,
+    BOOK_REQUIREMENTS_BEGINNER,
+    BOOK_SKILLRANGE_MAX,
+    BOOK_SKILLRANGE_MIN,
+    BOOK_REQUIREMENTS_INT,
+    BOOK_MORALECHANGE,
+    BOOK_TIMEPERCHAPTER,
+    BOOK_NUMUNREADCHAPTERS,
+
+    DESCRIPTION_BOOK_RECIPES,
+    DESCRIPTION_BOOK_ADDITIONAL_RECIPES,
+
+    BOOK_UNREAD,
+
+
+    CONTAINER_DETAILS,
+
+
+    TOOL_CHARGES,
+    TOOL_MAGAZINE_CURRENT,
+    TOOL_MAGAZINE_COMPATIBLE,
+    TOOL_CAPACITY,
+
+
+    DESCRIPTION_COMPONENTS_MADEFROM,
+    DESCRIPTION_COMPONENTS_DISASSEMBLE,
+
+
+    QUALITIES,
+    QUALITIES_CONTAINED,
+
+
+    DESCRIPTION,
+    DESCRIPTION_TECHNIQUES,
+    DESCRIPTION_GUNMOD_ADDREACHATTACK,
+    DESCRIPTION_MELEEDMG,
+    DESCRIPTION_MELEEDMG_CRIT,
+    DESCRIPTION_MELEEDMG_BASH,
+    DESCRIPTION_MELEEDMG_CUT,
+    DESCRIPTION_MELEEDMG_PIERCE,
+    DESCRIPTION_MELEEDMG_MOVES,
+    DESCRIPTION_APPLICABLEMARTIALARTS,
+    DESCRIPTION_REPAIREDWITH,
+
+    DESCRIPTION_CONDUCTIVITY,
+    DESCRIPTION_FLAGS,
+    DESCRIPTION_FLAGS_HELMETCOMPAT,
+    DESCRIPTION_FLAGS_FITS,
+    DESCRIPTION_FLAGS_VARSIZE,
+    DESCRIPTION_FLAGS_SIDED,
+    DESCRIPTION_FLAGS_POWERARMOR,
+    DESCRIPTION_FLAGS_POWERARMOR_RADIATIONHINT,
+    DESCRIPTION_IRRIDATION,
+
+    DESCRIPTION_RECHARGE_UPSMODDED,
+    DESCRIPTION_RECHARGE_NORELOAD,
+    DESCRIPTION_RECHARGE_UPSCAPABLE,
+
+    DESCRIPTION_RADIO_ACTIVATION,
+    DESCRIPTION_RADIO_ACTIVATION_CHANNEL,
+    DESCRIPTION_RADIO_ACTIVATION_PROC,
+
+    DESCRIPTION_CBM_SLOTS,
+
+    DESCRIPTION_TWOHANDED,
+    DESCRIPTION_GUNMOD_DISABLESSIGHTS,
+    DESCRIPTION_RADIOACTIVITY_DAMAGED,
+    DESCRIPTION_RADIOACTIVITY_ALWAYS,
+
+    DESCRIPTION_BREWABLE_DURATION,
+    DESCRIPTION_BREWABLE_PRODUCTS,
+
+    DESCRIPTION_FAULTS,
+
+    DESCRIPTION_HOLSTERS,
+
+    DESCRIPTION_ACTIVATABLE_TRANSFORMATION,
+
+    DESCRIPTION_NOTES,
+
+    DESCRIPTION_CONTENTS,
+
+    DESCRIPTION_APPLICABLE_RECIPES,
+
+
+    // misc 'block' range trackers
+    RANGE_TEXT_START = DESCRIPTION,
```

Comment:
These two could be defined as `static constexpr` (separate from the enumeration). That would allow to remove the explicit value from `MAX_VALUE` (being at the end, it would get the maximal value automatically, compare `NUM_AEPS` and similar `NUM_*` values).

Reply:
I remember reading somewhere that gcc emits warnings for scoped enums in switch statements if not all members have a case branch.

The `MAX_VALUE = [LAST_ENTRY]` approach squashes those warnings.

But ultimately, that's mere flavor.

As for moving the ranges externally... Yeah sure, but then things might get missed by a later maintainer. Keep the values inline and it's a lot harder to screw this up.